### PR TITLE
stanford-corenlp 3.9.1 (new formula)

### PR DESCRIPTION
**Pull request by DocLM**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/stanford-corenlp.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class StanfordCorenlp < Formula
+  desc "StanfordCoreNLP -- a suite of NLP tools"
```
Comment (ilovezfs) [inline Formula/stanford-corenlp.rb]: The description shouldn't start with `StanfordCoreNLP -- a `

File: Formula/stanford-corenlp.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  depends_on :java => "1.8+"
+
+  def install
+    libexec.install Dir["*"]
+    bin.write_exec_script Dir["#{libexec}/*.sh"]
+  end
+
+  test do
+    system "#{bin}/corenlp.sh", "-ner.useSUTime 0", "-file #{libexec}/input.txt"
```
Comment (ilovezfs) [inline Formula/stanford-corenlp.rb]: The test should assert something beyond just a zero exit code.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/stanford-corenlp.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class StanfordCorenlp < Formula
+  desc "StanfordCoreNLP -- a suite of NLP tools"
+  homepage "https://stanfordnlp.github.io/CoreNLP/"
+  url "https://nlp.stanford.edu/software/stanford-corenlp-full-2018-02-27.zip"
+  version "3.9.1"
```
[the remark this answers is not in the document]
Comment (DocLM) [reply Formula/stanford-corenlp.rb]: I found it on the project homepage here: https://stanfordnlp.github.io/CoreNLP/

File: Formula/stanford-corenlp.rb (hunk 2 above)
Comment (sbibauw) [reply Formula/stanford-corenlp.rb]: @DocLM Here is a proposal for test:
```ruby
 test do
  (testpath/"test.txt").write("Stanford is a university, founded in 1891.")
  system "#{bin}/corenlp.sh", "-annotators tokenize,ssplit,pos", "-file test.txt"
  assert_predicate "test.txt.xml", :exist?
```
It is more basic than the NER in your test, but runs much faster (1 min vs 37 min on my MBP).

File: Formula/stanford-corenlp.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class StanfordCorenlp < Formula
+  desc "A Java suite of core NLP tools "
```
Comment (sbibauw) [inline Formula/stanford-corenlp.rb]: To satisfy `brew audit`, you should change the description to:
```ruby
desc "Java suite of core NLP tools"
```

**Comment by claui** [comment]
@BrewTestBot test this please

**Review by claui — APPROVED** [review]

**Pull request merged**